content/docs/04-clusters/03-edge/04-edgeforge-workflow/03.5-palette-canvos.md

```diff
@@ -0,0 +1,658 @@
+  <br />
+
+  ```shell
+  git checkout v3.4.3
+  ```
+
+
+5. Review the files relevant for this guide. 
+    - **.arg** - Defines arguments to use during the build process. 
+    - **Dockerfile** - Bakes the arguments to use in the image.
+    - **Earthfile** - Contains a series of commands to create target artifacts.
+    - **earthly.sh** - Script to invoke the Earthfile, and generate target artifacts.
+    - **user-data.template** - A sample user-data file
+  <br />
+
+
+6. Open the **.arg** file in an editor to change one of the arguments. 
+<br />
+
+  ```bash
+  vi .arg
+  ```
+  <br />
+
+7. In the **.arg** file, change the value of the `CUSTOM_TAG` variable to any lowercase and alphanumeric string. This variable defines the custom image tag for the provider OS images. This guide uses the following custom tag as an example. 
+<br /> 
+
+  ```bash
+  CUSTOM_TAG=palette-learn
+  ```
+  Save the file, and exit the editor. 
+  <br />
+
+8. Issue the command below to save your registration token to a local variable. 
+<br />
+
+  ```bash
+  export token="your_token_here"
+  ```
+  <br />
+  
+9. Copy and issue the command below to create the **user-data** file that includes the tenant registration token. Also, you can click on the *Points of Interest* numbers below to learn more about the main attributes relevant to this example. 
+  <br />
+
+  <PointsOfInterest
+    points={[
+      {
+        x: 260,
+        y: 187,
+        label: 1,
+        description: "Stores the registration token and lets the agent use the auto-registration functionality and authenticate with the provided token.",
+        tooltipPlacement: "rightTop",
+      },
+      {
+        x: 190,
+        y: 262,
+        label: 2,
+        description: "Instructs the installer to turn the host machine off once the installation is complete.",
+      },
+      {
+        x: 190,
+        y: 340,
+        label: 3,
+        description: "Sets the login credentials for Edge hosts. The login credentials will allow you to SSH log into the edge host for debugging purposes.",
+        tooltipPlacement: "rightTop",
+      },
+    ]}
+  >
+
+  ```shell
+  cat << EOF > user-data
+  #cloud-config
+  stylus:
+    site:
+      edgeHostToken: $token
+  install:
+    poweroff: true
+  users:
+    - name: kairos
+      passwd: kairos
+  EOF
+  ```
+
+  </PointsOfInterest>
+  <br />
+
+
+10. CanvOS utility internally uses [Earthly](https://earthly.dev/) to build the target artifacts. Issue the following command to invoke the Earthly build. The `--PE_VERSION` option in the command below signifies the Palette Edge version to use, which refers to the git tag you checked out in the previous step. 
+<br />
+  <WarningBox>
+
+  Ensure to issue the command below with root-level or `sudo` privileges.
+
+  </WarningBox>
+
+  ```shell
+  ./earthly.sh +build-all-images --PE_VERSION=$(git describe --abbrev=0 --tags)
```

Comment:
```suggestion
  sudo ./earthly.sh +build-all-images --PE_VERSION=$(git describe --abbrev=0 --tags)
```
Let's just add the `sudo` and remove the warning box. Remember, people don't read, they will copy and paste.